src/scripting_api/aform.js

```diff
@@ -52,6 +53,94 @@ class AForm {
     return event.target ? `[ ${event.target.name} ]` : "";
   }
 
+  _tryToGuessDate(cFormat, cDate) {
+    // We use the format to know the order of day, month, year, ...
+    // Hence we've it we know how to interpret the numbers we've in the date.
+
+    let actions = this._dateActionsCache.get(cFormat);
+    if (!actions) {
+      actions = [];
+      this._dateActionsCache.set(cFormat, actions);
+      cFormat.replace(
+        /(d+)|(m+)|(y+)|(H+)|(M+)|(s+)/g,
+        function (match, d, m, y, H, M, s) {
+          if (d) {
+            actions.push((n, now) => {
+              if (n >= 1 && n <= 31) {
+                now.setDate(n);
+                return true;
+              }
+              return false;
+            });
+          } else if (m) {
+            actions.push((n, now) => {
+              if (n >= 1 && n <= 12) {
+                now.setMonth(n - 1);
+                return true;
+              }
+              return false;
+            });
+          } else if (y) {
+            actions.push((n, now) => {
+              if (n < 50) {
+                n += 2000;
+              } else if (n < 100) {
+                n += 1900;
+              }
+              now.setYear(n);
+              return true;
+            });
+          } else if (H) {
+            actions.push((n, now) => {
+              if (n >= 0 && n <= 23) {
+                now.setHours(n);
+                return true;
+              }
+              return false;
+            });
+          } else if (M) {
+            actions.push((n, now) => {
+              if (n >= 0 && n <= 59) {
+                now.setMinutes(n);
+                return true;
+              }
+              return false;
+            });
+          } else if (s) {
+            actions.push((n, now) => {
+              if (n >= 0 && n <= 59) {
+                now.setSeconds(n);
+                return true;
+              }
+              return false;
+            });
+          }
+          return "";
+        }
+      );
+    }
+
+    const number = /\d+/g;
+    let i = 0;
+    let array;
+    const now = new Date();
```

Comment:
This feels *a little* confusing as a variable name, since it may contain *any* date and not just "now"; how about the following instead? 
```suggestion
    const date = new Date();
```